sdk/servicebus/azure-servicebus/azure/servicebus/_common/message.py

```diff
@@ -581,17 +558,55 @@ def add_message(self, message):
         self._messages.append(message)
 
 
-class ServiceBusPeekedMessage(ServiceBusMessage):
-    """A preview message.
+class ServiceBusReceivedMessage(ServiceBusMessage):
+    """
+    A Service Bus Message received from service side.
+
+    :ivar auto_renew_error: Error when AutoLockRenewer is used and it fails to renew the message lock.
+    :vartype auto_renew_error: ~azure.servicebus.AutoLockRenewTimeout or ~azure.servicebus.AutoLockRenewFailed
+
+    .. admonition:: Example:
+
+        .. literalinclude:: ../samples/sync_samples/sample_code_servicebus.py
+            :start-after: [START receive_complex_message]
+            :end-before: [END receive_complex_message]
+            :language: python
+            :dedent: 4
+            :caption: Checking the properties on a received message.
 
-    This message is still on the queue, and unlocked.
-    A peeked message cannot be completed, abandoned, dead-lettered or deferred.
-    It has no lock token or expiry.
     """
+    def __init__(self, message, receive_mode=ReceiveMode.PeekLock, **kwargs):
+        # type: (uamqp.message.Message, ReceiveMode, Any) -> None
+        super(ServiceBusReceivedMessage, self).__init__(None, message=message)  # type: ignore
+        self._settled = (receive_mode == ReceiveMode.ReceiveAndDelete)
+        self._received_timestamp_utc = utc_now()
+        self._is_deferred_message = kwargs.get("is_deferred_message", False)
+        self._is_peeked_message = kwargs.get("is_peeked_message", False)
+        self.auto_renew_error = None  # type: Optional[Exception]
+        try:
+            self._receiver = kwargs.pop("receiver")  # type: Union[ServiceBusReceiver]
+        except KeyError:
+            raise TypeError("ReceivedMessage requires a receiver to be initialized.  This class should never be" +
+                            "initialized by a user; the Message class should be utilized instead.")
```

Comment:
A couple of tiny things:
- On line 587, the type hint has a Union of only one type
- In the error message - ReceivedMessage -> ServiceBusReceivedMessage, and Message too.
- I would probably expand the error message to say "for outgoing messages, the ServiceBusMessage class should be..."

Reply:
global search and replace, they should all align now :)